graphjet-core/src/test/java/com/twitter/graphjet/bipartite/edgepool/RegularDegreeEdgePoolTest.java

```diff
@@ -195,4 +201,67 @@ public void testRandomConcurrentReadWrites() {
     testRandomConcurrentReadWriteThreads(
         regularDegreeEdgePool, 3, 10 * maxNumNodes, maxDegree, 0.1, random);
   }
+
+  @Test
+  public void testGetNodeEdgeMetadataReturningNegative() {
+      NullStatsReceiver nullStatsReceiver = new NullStatsReceiver();
+      RegularDegreeEdgePool regularDegreeEdgePool = new RegularDegreeEdgePool(4, 4, nullStatsReceiver);
+      regularDegreeEdgePool.addEdge(2048, 587, (-424L));
+
+      assertEquals((-1L), regularDegreeEdgePool.getNodeEdgeMetadata(2048, 426));
+ }
+
+  @Test
+  public void testAddEdgeTakingThreeArgumentsThrowsIllegalArgumentException() {
+      NullStatsReceiver nullStatsReceiver = new NullStatsReceiver();
+      RegularDegreeEdgePool regularDegreeEdgePool = new RegularDegreeEdgePool(508, 2, nullStatsReceiver);
+      ShardedBigIntArray shardedBigIntArray = new ShardedBigIntArray(3157, 2, (-1035), nullStatsReceiver);
+      ShardedBigLongArray shardedBigLongArray = new ShardedBigLongArray(1431655765, 2, 2, nullStatsReceiver);
+      IntToIntPairArrayIndexBasedMap intToIntPairArrayIndexBasedMap = new IntToIntPairArrayIndexBasedMap(3157, 16, nullStatsReceiver);
+      RegularDegreeEdgePool.ReaderAccessibleInfo regularDegreeEdgePool_ReaderAccessibleInfo = new RegularDegreeEdgePool.ReaderAccessibleInfo(shardedBigIntArray, shardedBigLongArray, intToIntPairArrayIndexBasedMap);
+      regularDegreeEdgePool.readerAccessibleInfo = regularDegreeEdgePool_ReaderAccessibleInfo;
+
+      try {
+          regularDegreeEdgePool.addEdge(0, 0, (-2353L));
+          fail("Expecting exception: IllegalArgumentException");
+      } catch (IllegalArgumentException e) {
+          assertEquals(Preconditions.class.getName(), e.getStackTrace()[0].getClassName());
+      }
+  }
+
+  @Test
+  public void testFailsToCreateRegularDegreeEdgePoolTakingThreeArgumentsThrowsIllegalArgumentExceptionOne() {
```

Comment:
not a big fan of super long function names... When a test case fails, the test file name appends the failed test case name in the error message. in this case, it will say RegularDegreeEdgePoolTest.testFailsToCreateRegularDegreeEdgePoolTakingThreeArgumentsThrowsIllegalArgumentExceptionOne fails. Thus, there is little need to include the test file name in every test case name.
In this case, how about "testFailsTakingThreeArgumentsOne"? Similarly, a few other test case names are longer than necessary.